README.md

```diff
@@ -170,31 +170,59 @@ For `sycl_implementation` this can be one of:
 
 You can also specify the additional optional options:
 
+```
 -DSYCL_ACADEMY_INSTALL_ROOT=<path_to_sycl_impl_install_root>
+```
 
 For `<path_to_sycl_impl_install_root>` we recommend you specify the path to the
 root directory of your SYCL implementation installation, though this may not
 always be required.
 
+```
 -DSYCL_ACADEMY_ENABLE_SOLUTIONS=ON
+```
 
 This will enable building the solutions for each exercise as well as the source
 files. This is disabled by default.
 
+```
 -DCMAKE_BUILD_TYPE=Release
+```
 
 The build configuration for all exercises defaults to a debug build if this option is not specified.
 
 #### Additional cmake arguments for DPC++
 
+```
 -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx
+```
 
 This SYCL Academy CMake configuration uses the Intel oneAPI IntelSYCL CMake module package to assist it in its configuration. These command line arguments must be used to initiate this configuration correctly.
 
+`SYCL_TRIPLE` can be used to specify a DPC++ compatible SYCL triple. Possible
+values include:
+
+* `nvptx64-nvidia-cuda` - For CUDA devices
+* `spir64-gen` - For Intel GPUs
+* `native_cpu` - For native CPU SYCL device (dependent on DPCPP version)
+
+If AMD is being targeted, `AMD_ARCH` and `ROCM_DIR` must be set. Possible values
```

Comment:
Why is AMD treated differently from NVIDIA? They support identical interfaces. Both can be used with `-fsycl-targets=TARGET` and with `-Xsycl-target-backend --offload-arch=ARCH`. In both cases the `--cuda-path` / `--rocm-path` are optional and only needed if packages aren't installed system-wide or the right environment isn't set up.

Could you make both supported through `SYCL_TRIPLE` and add an option to specify ARCH for both? I would like it to be possible to specify e.g.:  
`-DSYCL_TRIPLE=nvptx64-nvidia-cuda -DSYCL_ARCH=sm_80`  
or  
`-DSYCL_TRIPLE=amdgcn-amd-amdhsa -DSYCL_ARCH=gfx90a`  

Optionally (if we want to support it - see my other comment) also:
`-DSYCL_TRIPLE=spir64_gen -DSYCL_ARCH=pvc` 
but that requires extra logic (different flags than NVIDIA/AMD).

Reply:
Thanks change made. It looks like it is generating the PVC compilation command that you suggest but it is failing for me at `oclc`. I'm not sure if this is because how I have configured my build. Would be great if you could check if the new pvc build command works for you

Reply:
I have unified the arch into `SYCL_ARCH` but still think it's good to use CMake vars `CUDA_DIR` and `ROCM_DIR`, as I think a generic var for this might sound confusing: `NATIVE_SDK_DIR` isn't really intuitive whereas the other ones are

Reply:
The pvc one fails for me with open-source DPC++ as well. It looks to be adding ocloc argument `-device_options pvc -ze-intel-enable-auto-large-GRF-mode` which doesn't happen in the 2024.0 release. My ocloc doesn't recognise `-device_options`, maybe it was added in a recent version.

Reply:
Yeah I'm not going to pretend I know what's happening there. I might leave this here anyway and we can see if it works further down the line. Or do you think I should remove references to `spir634_gen`?

Reply:
Fine with me either way. The PR is good and useful, and the new NVIDIA/AMD options work well, so I'm happy to approve this as it is.

Reply:
Great thanks @rafbiels